tempodb/backend/s3/config.go

```diff
@@ -87,6 +87,7 @@ func (cfg *Config) RegisterFlagsAndApplyDefaults(prefix string, f *flag.FlagSet)
 	f.StringVar(&cfg.SSE.KMSEncryptionContext, util.PrefixConfig(prefix, "s3.sse.kms-encryption-context"), "", "KMS Encryption Context used for object encryption. It expects JSON formatted string.")
 	f.StringVar(&cfg.SSE.CustomerEncryptionKey, util.PrefixConfig(prefix, "s3.sse.encryption-key"), "", "SSE-C Encryption Key used for object encryption.")
 	cfg.HedgeRequestsUpTo = 2
+	cfg.PartSize = 100 * 1024 * 1024 // 100MB default, matches parquet row group size (R2 compliant, min 5MB)
```

Comment:
Can we leave default at 0 and in this case it observes the old behavior? If the user sets this field then it observes the new behavior?

I'm concerned this will spike the API request rate for s3 users as written.